addons/mail/static/src/messaging/component/message/message_tests.js

```diff
@@ -347,6 +346,344 @@ QUnit.test('Notification Error', async function (assert) {
     );
 });
 
+QUnit.test("'channel_fetch' notification received is correctly handled", async function (assert) {
+    assert.expect(3);
+
+    await this.start();
+    const currentPartner = this.env.entities.Partner.create({
+        id: this.env.session.partner_id,
+        display_name: "Demo User"
+    });
+    const thread = this.env.entities.Thread.create({
+        id: 11,
+        members: [
+            [['link', currentPartner]],
+            [['insert', { id: 11, display_name: "Receiver" }]]
+        ],
+        model: 'mail.channel',
+    });
+    const threadViewer = this.env.entities.ThreadViewer.create({ thread: [['link', thread]] });
+    const message = this.env.entities.Message.create({
+        author: [['link', currentPartner]],
+        body: "<p>Test</p>",
+        id: 100,
+        threadCaches: [['link', thread.mainCache]],
+    });
+    await this.createMessageComponent(message, { threadViewerLocalId: threadViewer.localId });
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a message component"
+    );
+    assert.containsNone(
+        document.body,
+        '.o_MessageSeenIndicator_icon',
+        "message component should not have any check (V) as message is not yet received"
+    );
+
+    // Simulate received channel fetched notification
+    const notifications = [
+        [['myDB', 'mail.channel', 11], {
+            info: 'channel_fetched',
+            last_message_id: 100,
+            partner_id: 11,
+        }],
+    ];
+    await afterNextRender(() => {
+        this.widget.call('bus_service', 'trigger', 'notification', notifications);
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_MessageSeenIndicator_icon',
+        "message seen indicator component should only contain one check (V) as message is just received"
+    );
+});
+
+QUnit.test("'channel_seen' notification received is correctly handled", async function (assert) {
+    assert.expect(3);
+
+    await this.start();
+    const currentPartner = this.env.entities.Partner.create({
+        id: this.env.session.partner_id,
+        display_name: "Demo User"
+    });
+    const thread = this.env.entities.Thread.create({
+        id: 11,
+        members: [
+            [['link', currentPartner]],
+            [['insert', { id: 11, display_name: "Receiver" }]]
```

Comment:
In English we call the person at the other end "recipient", not "receiver". Receiver is used for objects... how rude!